product_pricelist_direct_print_company_group/wizards/product_pricelist_print.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2022 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, models
+
+
+class ProductPricelistPrintCompanyGroup(models.TransientModel):
+    _inherit = "product.pricelist.print"
+
+    @api.model
+    def _get_sale_order_domain(self, partner):
+        domain = super()._get_sale_order_domain(partner)
+        if partner.company_group_member_ids:
+            domain.remove(("partner_id", "child_of", partner.id))
```

Comment:
Just as an informative note, if remove doesn't find the value it throws an exception.

I'm considering if it is really useful to add this complexity...
Perhaps the "normal" thing would be to simply return the domain of the super or the one defined here depending on the condition.
If a module modifies the original domain, we will see what we want to do, because at this moment we cannot know if the result after that modification will be the one that interests us.

I also plead guilty :sweat_smile: 
https://github.com/OCA/contract/pull/797/files#diff-4cbf45e260f90f16486df50dd7c169ad2918d55b0abc9603f86c9298643f544aR13-R22

Reply:
Hmmm... IDK too

I corrected what I have made to avoid possible errors, but is true that If any module modifies the domain changing the partners domain the result can take a different direction...